packages/meditrak-server/src/tests/routes/exportSurveyResponses/exportSurveyResponses.test.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { expect } from 'chai';
+import sinon from 'sinon';
+import xlsx from 'xlsx';
+import { Authenticator } from '@tupaia/auth';
+import { buildAndInsertSurveyResponses, buildAndInsertSurveys } from '@tupaia/database';
+import { TUPAIA_ADMIN_PANEL_PERMISSION_GROUP } from '../../../permissions';
+import { TestableApp } from '../../TestableApp';
+import { INFO_COLUMN_HEADERS } from '../../../routes/exportSurveyResponses/exportSurveyResponses';
+
+const DEFAULT_POLICY = {
+  DL: ['Public'],
+  KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin', 'Public'],
+  SB: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Royal Australasian College of Surgeons'],
+  VU: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin', 'Donor'],
+  LA: ['Admin'],
+};
+
+const TEST_SURVEY_1_CODE = 'TEST_SURVEY_EXPORT_SURVEY_RESPONSES_1';
+const TEST_SURVEY_1_NAME = 'Export Survey SR 1';
+const TEST_SURVEY_2_CODE = 'TEST_SURVEY_EXPORT_SURVEY_RESPONSES_2';
+const TEST_SURVEY_2_NAME = 'Export Survey SR 2';
+
+const prepareStubAndAuthenticate = async (app, policy = DEFAULT_POLICY) => {
+  sinon.stub(Authenticator.prototype, 'getAccessPolicyForUser').returns(policy);
+  await app.authenticate();
+};
+
+const expectAccessibleExportDataHeaderRow = aoaExportData => {
+  expect(aoaExportData.length).to.be.greaterThan(1);
+  const headerRow = aoaExportData[0];
+
+  for (let i = 0; i < INFO_COLUMN_HEADERS.length; i++) {
+    const headerCell = headerRow[i];
+    expect(headerCell).to.be.equal(INFO_COLUMN_HEADERS[i]);
+  }
+};
+
+describe('exportSurveys(): GET export/surveysResponses', () => {
+  const app = new TestableApp();
+  const models = app.models;
+
+  describe('Test permissions when exporting survey responses', async () => {
+    let vanuatuCountry;
+    let survey1;
+    let survey2;
+
+    before(async () => {
+      sinon.stub(xlsx.utils, 'aoa_to_sheet');
+
+      const adminPermissionGroup = await models.permissionGroup.findOne({ name: 'Admin' });
+      const donorPermissionGroup = await models.permissionGroup.findOne({ name: 'Donor' });
```

Comment:
This depends on the `Admin`, `Donor` groups already existing in the database. It would be safer to use `findOrCreateDummyRecord`, since in the future we may decide to run our tests on an empty database (or change the Donor group, for example).

Not much of an issue this so adding this as an optional suggestion!

Reply:
Good point about using `findOrCreateDummyRecord` @kael89 - that would certainly make this test more independent. However, we do have several tests in `meditrak-server` that rely on `scaffolding.js` to setup some data as per `addBaselineTestData.js` - so Chris is safe here!

Reply:
Oh good point, I wasn't aware of that functionality!

But this does not apply to `Vanuatu`. In the meantime, @chris-bes confirmed that this indeed causes an issue since Vanuatu does not exist in the CI/CD environment when this test runs

Reply:
I've also just realized that in a few of my tests I was relying on some country entities to exist (eg: Vanuatu, Laos or Kiribati) which cause the tests to fail in the actual build (my bad for only check if the tests work locally). I will go back to fix those tests!